contracts/V1/StateChannelDiamondProxy/DisputeManagerFacet.sol

```diff
@@ -17,18 +17,16 @@ contract DisputeManagerFacet is StateChannelCommon {
         require(msg.sender == dispute.disputer, "CREATE DISPUTE: INVALID DISPUTER");
         // race condition check
         address[] memory onChainSlashedParticipants = getOnChainSlashedParticipants();
-        if(dispute.onChainSlashes.length != onChainSlashedParticipants.length 
-        || dispute.onChainSlashes[dispute.onChainSlashes.length - 1] != onChainSlashedParticipants[onChainSlashedParticipants.length - 1]) {
+
+        if(keccak256(abi.encodePacked(dispute.onChainSlashes)) != keccak256(abi.encodePacked(onChainSlashedParticipants))) {
```

Comment:
Please test this - I'm not sure this will compile since storage arrays don't work the same as memory arrays (they're essentially mappings), so I don't think you can abi.encodePacked(storage array)

Your original idea should also work, just add a check to not do [-1]